src/snowflake/snowpark/dataframe_writer.py

```diff
@@ -53,7 +66,363 @@ class DataFrameWriter:
+            self._copy_options[key.lower()] = value
+            return self
+        elif key.upper() in OPTION_ALIASES:
+            supported_key, convert_value_function = OPTION_ALIASES[key.upper()]
+            key = supported_key.upper()
+            value = convert_value_function(value)
+        self._cur_options[key.upper()] = value
+        return self
+
+    def options(self, configs: Dict) -> "DataFrameWriter":
+        """Sets multiple specified options in the DataFrameReader.
+
+        This method is the same as the :meth:`option` except that you can set multiple options in one call.
+
+        Args:
+            configs: Dictionary of the names of options (e.g. ``compression``,
+                ``skip_header``, etc.) and their corresponding values.
+        """
+        for k, v in configs.items():
+            self.option(k, v)
+        return self
+
+    def format(self, format: Literal["csv", "json", "parquet"]) -> "DataFrameWriter":
+        """
+        Sets the file type that will be use for unloading.
+
+        This method allows specifying the file type that will used for data unloading. The accepted file formats are:
+        - csv
+        - json
+        - parquet
+
+        Alternatives to using the `format` method are:
+        1. Using the `option` method with either "format" or "type" key-value pairs.
+        Example: option("format", "csv") or option("type", "json")
+
+        2. Passing the file type directly into the `load` method.
+        Example: load("path/to/your/file", format="json")
+
+        Note:
+            Ensure that you use a valid file format and use the appropriate method or option to specify it.
+
+        Args:
+            format: The file format to use for ingestion.
+
+        Returns:
+            DataFrameReader: The DataFrameReader object with the specified format set.
+        """
+        self._file_format_type = format
+        return self
+
+    @overload
+    def copy_into_location(
+        self,
+        location: str,
+        *,
+        partition_by: Optional[ColumnOrSqlExpr] = None,
+        file_format_name: Optional[str] = None,
+        file_format_type: Optional[str] = None,
+        format_type_options: Optional[Dict[str, str]] = None,
+        header: bool = False,
+        statement_params: Optional[Dict[str, str]] = None,
+        block: bool = True,
+        **copy_options: Optional[str],
+    ) -> List[Row]:
+        ...  # pragma: no cover
+
+    @overload
+    def copy_into_location(
+        self,
+        location: str,
+        *,
+        partition_by: Optional[ColumnOrSqlExpr] = None,
+        file_format_name: Optional[str] = None,
+        file_format_type: Optional[str] = None,
+        format_type_options: Optional[Dict[str, str]] = None,
+        header: bool = False,
+        statement_params: Optional[Dict[str, str]] = None,
+        block: bool = True,
+        **copy_options: Optional[str],
+    ) -> List[Row]:
+        ...  # pragma: no cover
```

Comment:
this is added twice. See lines 395-425 below.

Also, can this be moved below next to other overloaded `copy_into_location` definitions?